src/com/jwetherell/algorithms/mathematics/Division.java

```diff
@@ -55,6 +67,9 @@ public static final long divisionUsingMultiplication(int a, int b) {
     }
 
     public static final long divisionUsingShift(int a, int b) {
+        if (b == 0) {
```

Comment:
Added an if statement that checks if the b integer equals zero. If it does the program will throw an IllegalArgumentException and then display an error message.